### PR TITLE
resolve green ci

### LINKED ISSUE #1281: [CRAN Release]: 0.9.6
#### PRs
- [ ] PR 1

#### Issues
- [ ] Issue 1

### Pre-release

- [ ] Make sure you adhere to CRAN submission policy:
  * https://cran.r-project.org/web/packages/submission_checklist.html
  * https://cran.r-project.org/web/packages/policies.html.
- [ ] Make sure that high priority bugs (label "priority" + "bug") have been resolved before going into the release.
- [ ] Review old/hanging PRs before going into the release (Optional).
- [ ] Revisit R-package's lifecycle badges (Optional).
- [ ] Make sure that all upstream dependencies of this package that need to be submitted to CRAN were accepted before going into release activities.
- [ ] Make sure integration tests are green 2-3 days before the release. Look carefully through logs (check for warnings and notes).
- [ ] Decide what gets merged in before starting release activities.

#### Prepare the release
- [ ] Create a new release candidate branch
  `git checkout -b release-candidate-vX.Y.Z`
- [ ] Update NEWS.md file: make sure it reflects a holistic summary of what has changed in the package.
- [ ] Remove the additional fields (`Remotes`) from the DESCRIPTION file where applicable.
- [ ] Make sure that the minimum dependency versions are updated in the DESCRIPTION file for the package and its reverse dependencies (Optional).
  - [ ] Increase versioned dependency on {package name} to >=X.Y.Z (Optional).
- [ ] Commit your changes and create the PR on GitHub (add "[skip vbump]" in the PR title). Add all updates, commit, and push changes:
  ```r
  # Make the necessary modifications to your files
  # Stage the changes
  git add <files your modified>
  # Commit the changes
  git commit -m "[skip vbump] <Your commit message>"
  git push origin release-candidate-vX.Y.Z`
  ```

#### Test the release
- [ ] Execute the manual tests on Shiny apps that are deployed on various hosting providers (Posit connect and shinyapps.io) - track the results in GitHub issue (Applicable only for frameworks that use Shiny).
- [ ] Monitor integration tests, if integration fails, create priority issues on the board.
- [ ] Execute UAT tests (Optional).

#### CRAN submission
- [ ] Tag the update(s) as a release candidate vX.Y.Z-rc<iteration-number> (e.g. v0.5.3-rc1) on the release candidate branch (release-candidate-vX.Y.Z).
  ```r
  # Create rc tag for submission for internal validation
  git tag vX.Y.Z-rc<iteration number>
  git push origin vX.Y.Z-rc<iteration number>
  ```
- [ ] Build the package locally using the command:`R CMD build .` which will generate a .tar.gz file necessary for the CRAN submission.
- [ ] Submit the package to https://win-builder.r-project.org/upload.aspx for testing, for more details please see "Building and checking R source packages for Windows": https://win-builder.r-project.org/.
- [ ] Once tested, send the package that was built in the previous steps to CRAN via this form: https://cran.r-project.org/submit.html.
- [ ] Address CRAN feedback, tag the package vX.Y.Z-rc(n+1) and repeat the submission to CRAN whenever necessary.
- [ ] Get the package accepted and published on CRAN.

#### Tag the release
- [ ] If the additional fields were removed, add them back in a separate PR, and then merge the PR back to main (add "[skip vbump]" in the PR title). If nothing was removed just merge the PR you created in the "Prepare the release" section to 'main'. Note the commit hash of the merged commit. **Note:** additional commits might be added to the `main` branch by a bot or an automation - we do **NOT** want to tag this commit.

##### Make sure of the following before continuing
- [ ] CI checks are passing in GH before releasing the package.
- [ ] Shiny apps are deployable and there are no errors/warnings (Applicable only for frameworks that use Shiny).

- [ ] Create a git tag with the final version set to vX.Y.Z on the main branch. In order to do this:
  1. Checkout the commit hash.
  `git checkout <commit hash>`
  2. Tag the hash with the release version (vX.Y.Z).
  `git tag vX.Y.Z`
  3. Push the tag to make the final release.
  `git push origin vX.Y.Z`
- [ ] Update downstream package dependencies to (>=X.Y.Z) in {package name}.
**Note:** Once the release tag is created, the package is automatically published to internal repositories.

### Post-release

- [ ] Ensure that CRAN checks are passing for the package.
- [ ] Make sure that the package is published to internal repositories.
- [ ] Make sure internal documentation is up to date.
- [ ] Review and update installation instructions for the package wherever needed (Optional).
- [ ] Announce the release on ________.

### Decision tree

Click [here](https://github.com/insightsengineering/.github/blob/main/.github/ISSUE_TEMPLATE/RELEASE_DECISION_TREE.md) to see the release decision tree.

### LINKED ISSUE #1282: CI green for https://github.com/pawelru/ie_status?tab=readme-ov-file
todo

 - [ ] dependencies test release https://github.com/insightsengineering/tern/actions/runs/10437589592/job/28903939454
 - [ ] dependencies test max
 - [ ] revdepcheck https://github.com/insightsengineering/tern/actions/runs/10437589592/job/28903939250
 - [ ] nosuggests https://github.com/insightsengineering/tern/actions/runs/10437589592/job/28903942900

### PR DESCRIPTION
**Pull request by shajoezhu**
close #1282

### CODE REVIEW

**Review by Melkiades — APPROVED** [review]
lgtm! Just need to fix url ;)

File: tests/testthat/test-individual_patient_plot.R [hunk 1]
```diff
@@ -3,6 +3,9 @@ adlb <- tern_ex_adlb %>%
   slice(1:36)
 
 testthat::test_that("h_g_ipp works correctly", {
+  require("nestcolor", quietly = TRUE)
+  skip_if_not_installed("nestcolor")
+
```
Comment (pawelru) [inline tests/testthat/test-individual_patient_plot.R:8]: - skip first before require
- I think it's fairly easy to remove dependency of `nestcolor` from this unit test code - just do `col = c("blue", "red", ...)` making sure that the length is the same

File: vignettes/tern.Rmd [hunk 2]
```diff
@@ -69,7 +69,7 @@ data visualizations helper functions:
 - ...
 
 
-The reference of `tern` functions is available on [the tern website functions reference](https://insightsengineering.github.io/tern/main/reference/index.html).
+The reference of `tern` functions is available on [the tern website functions reference](https://insightsengineering.github.io/tern/latest-tag/reference/index.html).
```
Comment (Melkiades) [inline vignettes/tern.Rmd:72]: I think this is only for {rtables} isnt it @pawelru?

Comment (pawelru) [reply vignettes/tern.Rmd:72]: I don't understand the question.

That change is good. We should refer to `/atest-tag/` wherever possible. I'm aware of only one exception to that rule. 
https://insightsengineering.github.io/rtables/main/articles/rtables.html
which does not exists for latest tag: https://insightsengineering.github.io/rtables/latest-tag/articles/rtables.html
in latest-tag its available under a different name: https://insightsengineering.github.io/rtables/latest-tag/articles/introduction.html

**Comment by shajoezhu** [comment]
I don't really want to drop "nestcolor::theme_nest()" from the g_lineplot default option, reason being this is primarily used by roche, and the nest default values are set for a reason, 

I don't want to replace the usage of nestcolor either, feel like this is going backwards, and adding code that we have to maintain in tern ...

one possibility is to move nestcolor from suggests to import @pawelru @khatril , what do you guys think?

**Comment by pawelru** [comment]
so my comment was only about individual unit test
I'm totally with you when it comes to the examples and the functionality as a whole

File: tests/testthat/test-individual_patient_plot.R [hunk 3]
```diff
@@ -3,6 +3,9 @@ adlb <- tern_ex_adlb %>%
   slice(1:36)
 
 testthat::test_that("h_g_ipp works correctly", {
+  require("nestcolor", quietly = TRUE)
+  skip_if_not_installed("nestcolor")
```
Comment (shajoezhu) [inline tests/testthat/test-individual_patient_plot.R:7]:
```suggestion
  skip_if_not_installed("nestcolor")
  require("nestcolor", quietly = TRUE)
```

**Comment by shajoezhu** [comment]
block this PR by https://github.com/insightsengineering/tern/pull/1311

**Comment by shajoezhu** [comment]
let's merge this in, and come back to resolve the green ci issue in future PRs

**Pull request merged**